querqy-for-lucene/querqy-solr/src/main/java/querqy/solr/IndexRewriterContainer.java

```diff
@@ -0,0 +1,493 @@
+
+        try {
+            withConfigurationCore(core -> core.withSearcher(this::loadRewriters), Duration.ofMinutes(1));
+        } catch (IOException e) {
+            throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, "Could not load rewriter data", e);
+        }
+
+        final Optional<SolrRequestHandler> requestHandler = Optional.ofNullable(core.getRequestHandler(replicationHandlerName));
+        requestHandler.ifPresent(handler -> {
+            if (handler instanceof ReplicationHandler) {
+                final var replicationHandler = (ReplicationHandler) handler;
+                this.isFollower = replicationHandler.isFollower();
+                LOGGER.warn("Querqy rewriter container is running in mode: {}", isFollower ? "follower" : "leader");
+            } else {
+                LOGGER.warn("Request handler '{}' is not an instance of solr.ReplicationHandler. Cannot detect if rewriter is leader or follower", replicationHandlerName);
+            }
+        });
+
+        if (isFollower) {
+            LOGGER.info("Initiate querqy configuration core polling in core {} with interval {}", core.getName(), pollingInterval);
+            final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+            scheduler.scheduleWithFixedDelay(this::checkAndReloadRewriterConfig, 0,  pollingInterval.getSeconds(), TimeUnit.SECONDS);
+            core.addCloseHook(new CloseHook() {
+                @Override
+                public void preClose(SolrCore core) {
+                    scheduler.shutdown();
+                }
+            });
+        }
+    }
+
+    @Override
+    protected void doClose() {
+        // nothing to close
+    }
+
+    private synchronized List<String> loadRewriters(final SolrIndexSearcher configurationSearcher) throws IOException {
+        generation = configurationSearcher.getIndexReader().getIndexCommit().getGeneration();
+        final List<String> storedRewriters = listStoredRewriterIds(configurationSearcher);
+        for (final String rewriterId : storedRewriters) {
+            loadRewriter(rewriterId, readRewriterDefinition(rewriterId, configurationSearcher));
+        }
+        return storedRewriters;
+    }
+
+    private void checkAndReloadRewriterConfig() {
+        LOGGER.info("Checking and reloading rewriter config for core {}", core.getName());
+        try {
+            withConfigurationCore(core -> core.withSearcher(searcher -> {
+                long currentGeneration = searcher.getIndexReader().getIndexCommit().getGeneration();
+                LOGGER.info("Querqy configuration index generation is {}. Last seen generation was {}.", currentGeneration, generation);
+                if (currentGeneration != generation) {
+                    reloadRewriterConfig(searcher);
+                }
+                return null;
+            }));
+        } catch (IOException e) {
```

Comment:
It is probably better to catch all Exceptions here. From what I remember RuntimeExceptions from Runnables are "swallowed" (not propagated to the scheduling main thread) and will prevent the Runnable from being re-scheduled. If we don't even log the Exception that is hard to detect.

Reply:
addressed in 41ec25747866a353046ebb147f16d9ce68568a04